packages/dashboard/src/portal/components/FarmNodesTable.vue

```diff
@@ -629,6 +686,27 @@ export default class FarmNodesTable extends Vue {
     }
     this.openPublicConfigDialog = true;
   }
+
+  async openExtraFee(node: nodeInterface) {
+    this.nodeToEdit = node;
+    // convert fees from USD to mili USD while getting
+    this.extraFee =
+      (await this.queryClient.contracts.getDedicatedNodeExtraFee({ nodeId: this.nodeToEdit.nodeId })) / 1000;
+    this.openExtraFeeDialogue = true;
+  }
+
+  async saveExtraFee(fee: number) {
+    this.loadingExtraFee = true;
+    // convert fees from mili USD to USD while setting
+    setDedicatedNodeExtraFee(this.$store.state.credentials.account.address, this.nodeToEdit.nodeId, fee * 1000).then(
+      () => {
+        this.$toasted.show(`Transaction succeeded: Fee is added to node ${this.nodeToEdit.nodeId}`);
+        this.loadingExtraFee = false;
+        this.openExtraFeeDialogue = false;
```

Comment:
can we also handle the failure case?
it keeps loading forever
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/41957921/8d209545-33ec-46c6-864f-dce7b1b545ad)

Reply:
handled.